src/unix/mmap.c

```diff
@@ -516,11 +516,9 @@ static void vmap_paint(heap h, rangemap pvmap, vmap q)
     assert((rq.end & MASK(PAGELOG)) == 0);
     assert(range_span(rq) > 0);
 
-    rmnode_handler nh = stack_closure(vmap_paint_intersection, h, pvmap, q);
-    rangemap_range_lookup(pvmap, rq, nh);
-
-    range_handler rh = stack_closure(vmap_paint_gap, h, pvmap, q);
-    rangemap_range_find_gaps(pvmap, rq, rh);
+    rangemap_range_lookup_with_gaps(pvmap, rq,
+                                    stack_closure(vmap_paint_intersection, h, pvmap, q),
+                                    stack_closure(vmap_paint_gap, h, pvmap, q));
 
```

Comment:
This doesn't work if the new mapping is a sub-range of an existing mapping and the existing mapping needs to be resized and moved at the end of the new mapping: in this case, the gap that is created by the resize is not caught by rangemap_range_lookup_with_gaps(), and as a result the gap handler is not called and the new mapping is not inserted in the vmap.

Reply:
Yes, thanks. Fixed.